webapp/src/main/java/com/box/l10n/mojito/service/cache/DatabaseCacheConfiguration.java

```diff
@@ -0,0 +1,93 @@
+package com.box.l10n.mojito.service.cache;
+
+import com.google.common.base.Preconditions;
+import org.springframework.beans.factory.annotation.Lookup;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Scope;
+
+import java.time.Duration;
+
+/**
+ * This class enables custom configurations for a database-backed cache.
+ * By default, no TTL is configured (an expiry date isn't writen for entries)
+ * and cache evictions for expired entries are attempted once every 10,000 writes.
```

Comment:
not sure to understand the eviction based on write, specifically for current use case, we'll never reach this number of writes?

Reply:
We can certainly configure that ratio / number to something we think is more appropriate to our use case.

The goal is to have a cache that takes care of its own clean-up/eviction - and doesn't require additional setup (e.g. external scheduled tasks) for clearing up expired entries - as that part will likely be missed and will result in stale data never getting cleaned.

There might a better mechanic out there to achieve this.
The way that goal is achieved in the current implementation is by scheduling an async cleanup of expired entries probabilistically every x writes (as the number of entries to be cleaned is generally proportional to the number of total entries - which is in turn proportional to the number of writes we do in the system).

Can you let me know if you can think of some better ways to achieve the same goal, while also keeping the cache self-contained?

Reply:
i can see how the self contained clean is better for in-memory cache but since this is a distributed cache assuming many clients, they'll try to do perform clean concurrently which does not seem great. So we'd need some kind of external scheduling system that prevent concurrent cleanups. We can do that with Quartz (`@DisallowConcurrentExecution` jobs): we have the capability to run 1 single task in the whole cluster. Still this can be trigger within the write based on the ratio or with a cron job.

Reply:
Using a Quartz job for the cleanup now.